src/h5web/dataset-visualizer/DatasetVisualizer.tsx

```diff
@@ -17,11 +17,23 @@ interface Props {
   dataset?: HDF5Dataset;
 }
 
-function DatasetVisualizer(props: Props): JSX.Element {
+function DatasetVisualizer(props: Props): ReactElement {
   const { dataset } = props;
 
-  const supportedVis = getSupportedVis(dataset);
-  const [activeVis, setActiveVis] = useActiveVis(supportedVis);
+  const supportedVis = useMemo(() => getSupportedVis(dataset), [dataset]);
```

Comment:
I re-added `useMemo` here after all so the method doesn't get called twice on every change of dataset (because of `setActiveVis`).